src/ophyd_async/epics/sim/_mover.py

```diff
@@ -52,9 +52,11 @@ async def set(self, value: float, timeout: CalculatableTimeout = CALCULATE_TIMEO
             self.precision.get_value(),
             self.velocity.get_value(),
         )
-        if timeout == CALCULATE_TIMEOUT:
-            assert velocity > 0, "Mover has zero velocity"
-            timeout = abs(new_position - old_position) / velocity + DEFAULT_TIMEOUT
+        if timeout is CALCULATE_TIMEOUT and velocity != 0:
+            timeout = abs((new_position - old_position) / velocity) + DEFAULT_TIMEOUT
+        else:
+            msg = "Mover has zero velocity"
```

Comment:
ditto

Reply:
fixed in 9d8828168188fff0aa43db025d17ac6da05a97f1
I'm not sure how I can test this, though.